File: array/array2d/go/array2d.go
package arrays

import "errors"

type Matrix struct {
	data [][]int
}

func NewMatrix(row_count, col_count int) *Matrix {
	matrix := make([][]int, row_count)
	for i := range matrix {
		matrix[i] = make([]int, col_count)
	}
	return &Matrix{
		data: matrix,
	}
}

func (m *Matrix) Get(row, col int) (int, error) {
	if !m.exists(row, col) {
		return 0, errors.New("index out of range")
	}
	return m.data[row][col], nil
}

func (m *Matrix) Set(row, col, value int) (int, error) {
	if !m.exists(row, col) {
		return 0, errors.New("index out of range")
	}
	m.data[row][col] = value
	return value, nil
}

func (m *Matrix) exists(row, col int) bool {
	return row >= 0 && row < len(m.data) && col >= 0 && col < len(m.data[row])
}

File: array/array2d/python/array2d.py
class Matrix:
    def __init__(self, row_count: int, col_count: int, default: int = 0) -> "Matrix":
        self.data = [[default] * col_count for _ in range(row_count)]

    def __repr__(self):
        return f"Matrix({self.data})"

    def get(self, row: int, col: int) -> int | ValueError:
        if not self._exists(row, col):
            raise ValueError("index out of range")
        return self.data[row][col]

    def set(self, row: int, col: int, value: int) -> int | ValueError:
        if not self._exists(row, col):
            raise ValueError("index out of range")
        self.data[row][col] = value
        return value

    def _exists(self, row: int, col: int) -> bool:
        return 0 <= row < len(self.data) and 0 <= col < len(self.data[row])
